src/main/scala/wdl4s/expression/WdlStandardLibraryFunctions.scala

```diff
@@ -329,6 +344,11 @@ class WdlStandardLibraryFunctionsType extends WdlFunctions[WdlType] {
   }
   def sub(params: Seq[Try[WdlType]]): Try[WdlType] = Success(WdlStringType)
   def range(params: Seq[Try[WdlType]]): Try[WdlType] = Success(WdlArrayType(WdlIntegerType))
+  def basename(params: Seq[Try[WdlValue]]): Try[WdlType] = params.toList match {
+    case Success(f) :: Nil if WdlStringType.isCoerceableFrom(f.wdlType) => Success(WdlStringType)
+    case Success(f) :: Success(s) :: Nil if WdlStringType.isCoerceableFrom(f.wdlType) && WdlStringType.isCoerceableFrom(s.wdlType) => Success(WdlStringType)
```

Comment:
Can we use this matching in the implementation above too to get `firstArgument` and `secondArgument` ?